src/main/kotlin/no/nav/syfo/personoppgavehendelse/PublishPersonoppgavehendelseService.kt

```diff
@@ -1,15 +1,46 @@
 package no.nav.syfo.personoppgavehendelse
 
-import no.nav.syfo.database.DatabaseInterface
-import no.nav.syfo.personoppgave.domain.PersonOppgave
+import no.nav.syfo.personoppgave.*
+import no.nav.syfo.personoppgave.domain.*
+import org.slf4j.LoggerFactory
+import java.sql.Connection
 
 class PublishPersonoppgavehendelseService(
-    private val database: DatabaseInterface,
     private val personoppgavehendelseProducer: PersonoppgavehendelseProducer,
 ) {
-    fun getUnpublishedOppgaver(): List<PersonOppgave> {
-        return emptyList()
+    private val log = LoggerFactory.getLogger(PublishPersonoppgavehendelseService::class.java)
+
+    fun getUnpublishedOppgaver(connection: Connection): List<PersonOppgave> {
+        return connection.getPersonOppgaverByPublish(publish = true).toPersonOppgaver()
     }
 
-    fun publish(personOppgave: PersonOppgave) {}
+    private fun isNewest(connection: Connection, personOppgave: PersonOppgave): Boolean {
+        val personOppgaver = connection.getPersonOppgaver(personOppgave.personIdent).toPersonOppgaver()
```

Comment:
Denne spørringen og mappingen kan vel potensielt bli kjørt flere ganger for samme person (dersom en person har flere unpublished oppgaver). Vet ikke om det er en issue, og mulig databasen optimaliserer det, men man kan jo se på om det går an å skrive om koden så man unngår det.